### ISSUE TITLE
TypeNameCheck doesn't work for Enum Constants

### ISSUE THREAD

**Issue by cristalp** [checkstyle/checkstyle#1110]
I want to prevent that Enums contain special characters like umlauts.

While check an enum class name like WillNötWörk will result in errors, an enum constant like WILL_NÖT_WÖRK is ignored.

I tried with the Maven Checkstyle plugin 2.15, which uses 6.1.1 and also tried overriding its version to the latest Checkstyle 6.6.
##

**Comment by romani** [comment]
checkstyle does not support parsing of non acsii sources, mostlikely parsing is failed so Ast subtree is skipped or smth else .

Please recheck that by CLI on latest release http://checkstyle.sourceforge.net/cmdline.html and share an output.

**Comment by cristalp** [comment]
Ok, I have the following three Enums:

```java
public enum Bar {

  FOO,

  BAR;
}
```

```java
public enum Foo {

  FÖÖ,

  BAR;
}
```

```java
public enum Föö {

  FÖÖ,

  BAR;
}
```
And this is the output when I test it with the Google Checks (which contain TypeName):

    $ java -jar checkstyle-6.6-all.jar -c /google_checks.xml ./Bar.java
    Starting audit...
    Audit done.
    $ java -jar checkstyle-6.6-all.jar -c /google_checks.xml ./Foo.java
    Starting audit...
    Audit done.
    $ java -jar checkstyle-6.6-all.jar -c /google_checks.xml ./Föö.java
    Starting audit...
    C:\jp\.\Föö.java:3:13: warning: Type name 'Föö' must match pattern '^[A-Z][a-zA-Z0-9]*$'.
    Audit done.

The class name with umlauts is handles correctly, but the enum constant with umlauts is ignored.

**Comment by rnveach** [comment]
@cristalp 
You are expecting "FÖÖ" of "Foo", and "FÖÖ" of "Föö" to be signaled as the wrong naming scheme, correct?

TypeNameCheck only checks the class level name, not the constants inside the class.
That is why it is only picking up "Föö".

@romani 
None of the other naming checks seem to handle "ENUM_CONSTANT_DEF" which I believe is the definition of the enumerated item.

``````
public enum Bar {
FOO,// <-- these guys
BAR;// <-- these guys
}
``````

I believe it should be added to "ConstantNameCheck" since it is static final but not written that way.

**Comment by cristalp** [comment]
@rnveach 
Yes, and I'm also expecting the Enum class name "Föö" to be incorrect, which works.

**Comment by mkordas** [comment]
I'd like to have separate check for that. Default naming convention should
be the same as for constants, but there are some projects that allow e.g.
lowercase enum literals because they use default valueOf and toString
heavily, so we should provide flexibility to users.

**Comment by romani** [comment]
Looks like it is already implemented https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/naming/EnumValueNameCheck.java

But, we are in moratorium period for new Checks, I remember that smth was not clear in that Check - https://groups.google.com/forum/#!searchin/sevntu-checkstyle/pavel/sevntu-checkstyle/c6FvyjXFcHM/cWsvTt4DV9QJ

@pbaranchikov , can you remind us a status of your testing on real code ? Do you have any thoughts for my last post in mail-list?

**Comment by pbaranchikov** [comment]
On real code we have not really used enum objects vs enum constants distinguishing. So, we successfully used the check, setting the same regexp for both fields. Other aspects look fine.

**Comment by romani** [comment]
issue is covered by sevntu Check - EnumValueNameCheck
